### PR TITLE
Gitlab: Optimizations for todo, issues, merge requests and CI pipelines/jobs

### PR DESCRIPTION
**Pull request by tonka3000**
## Description

- Fix possible `undefined` states for details view for MRs and Issues when open via todo command
- Closing a todo from the list will refresh/refetch list
- Closing or reopen an issue will refresh/refetch list
- Closing or reopen a MR will refresh/refetch list
- Set minimum Raycast version to `1.26.0`
- Add missing action icons
- CI Pipelines and CI Job view now auto refresh every 10 seconds by default

  Users can adjust the setting or fully deactivate it by setting the value to `0`. `10` is the same default as the GitLab website or the GitLab VSCode extension refresh interval. There is also a new `Refresh` action if a user wanna trigger the refresh manually.

  CleanShot 2022-01-23 at 15 57 31

  CleanShot 2022-01-23 at 16 00 14

## Type of change

- Bug fix / improvement for my extension

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

### CODE REVIEW

**Comment by tonka3000** [comment]
Review request 😀 @mattisssa @pernielsentikaer

File: extensions/gitlab/package.json [hunk 1]
```diff
@@ -145,17 +145,27 @@
       "description": "Your GitLab API token. `api` scope for read and write operations, `read_api` when you just want to read only commands.",
       "link": "https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html",
       "placeholder": "Enter your GitLab API token."
+    },
+    {
+      "name" : "cirefreshinterval",
+      "type" :"textfield",
+      "required": false,
+      "default": "",
```
[the remark this answers is not in the document]
Comment (tonka3000) [reply extensions/gitlab/package.json:153]: @mathieudutour Short question: Does raycast store the default value on the first use of the extension or only when the user change it manually. Just would love to change the default without anybody stuck on the old default when it was never changed by the user.

[the remark this answers is not in the document]
Comment (mathieudutour) [reply extensions/gitlab/package.json:153]: it doesn't store the default. When the user hasn't set anything as the value of the preference, `getPreferenceValues` will default to the default

[the remark this answers is not in the document]
Comment (tonka3000) [reply extensions/gitlab/package.json:153]: @mathieudutour Thanks for clarification

File: extensions/gitlab/src/common.ts [hunk 2]
```diff
@@ -48,3 +48,22 @@ export function createGitLabGQLClient() {
 
 export const gitlab = createGitLabClient();
 export const gitlabgql = createGitLabGQLClient();
+
+const defaultRefreshInterval = 10 * 1000;
+
+export function getCIRefreshInterval(): number | null {
+  const userValue = (preferences.cirefreshinterval?.value as string) || "10";
```
[the remark this answers is not in the document]
Comment (tonka3000) [reply extensions/gitlab/src/common.ts]: Thanks for the hint

**Review by mathieudutour — APPROVED** [review]
Looking good :+1:

**Pull request merged**